astroid/inference.py

```diff
@@ -185,7 +189,7 @@ def _infer_map(
 nodes.Dict._infer = infer_map  # type: ignore[assignment]
 
 
-def _higher_function_scope(node):
+def _higher_function_scope(node: nodes.LocalsDictNodeNG) -> nodes.FunctionDef | None:
```

Comment:
Why not annotate `node` itself with `NodeNG`? No attribute from `LocalsDictNodeNG` is used here.